### PR TITLE
Bug 1873043: Add rebase instructions

### PR DESCRIPTION
**Pull request by marun**
Add rebase instructions updated to reflect the transition of hyperkube maintenance from origin to openshift/kubernetes. For releases 4.6 and above, this document supercedes the instructions in the the [rebase enhancment](https://github.com/openshift/enhancements/blob/master/enhancements/rebase.md). Putting this document in the repository to which it applies should simplify its maintenance as part of the regular rebase workflow and also allow it to be more easily discovered.

A related PR updates the rebase enhancement to reflect its relevance to only releases 4.5 and below: https://github.com/openshift/enhancements/pull/481

```
Co-authored-by: Lukasz Szaszkiewicz <lukasz.szaszkiewicz@gmail.com>
```

### CODE REVIEW

File: REBASE.openshift.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+   (https://blog.nilbus.com/take-the-pain-out-of-git-conflict-resolution-use-diff3/)
+- Teach Git to remember how you’ve resolved a conflict so that the next time it can
+  resolve it automatically (https://git-scm.com/book/en/v2/Git-Tools-Rerere)
+
+## Preparing the local repo clone
+
+Clone from a personal fork of kubernetes:
+
+```
+git clone https://github.com/<user id>/kubernetes
```
Comment (tnozicka) [inline REBASE.openshift.md:30]: nit: I think you can just merge the 2 steps and always clone your own repo with `git@`

File: REBASE.openshift.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+```
+git checkout -b rebase-1.20.0 v1.20.0
+```
+
+- Merge the targeted `openshift/kubernetes` branch (e.g. `master`) with
+  strategy `ours` to reset the the branch to the targeted release tag without
+  involving manual conflict resolution.
+
+```
+git merge -s ours openshift/master
```
Comment (tnozicka) [inline REBASE.openshift.md:65]: My initial though is to do a rebase here, solve the conflicts and then compare the patches with the spreadsheet to be sure. I'll get back here with feedback when I actually try it.

Comment (p0lyn0mial) [reply REBASE.openshift.md:65]: yeah, I would have thought the same.

Comment (tnozicka) [reply REBASE.openshift.md:65]: I think we need to merge or we create different upstream commits in the history that won't match on the next rebase

Comment (tnozicka) [reply REBASE.openshift.md:65]: but shouldn't we merge the other way? at some point we've merged k-master into ocp-master, so we want to start from ocp-master and merge updated k-master into it to just get new commit from v1.19.0..v1.19.2 where v1.19.0 is the common merge base

Comment (tnozicka) [reply REBASE.openshift.md:65]: namely:
```
git fetch upstream --tags

git checkout -b ocp-rebase-1.19.2 openshift/master --no-track

git merge v1.19.2
```

File: REBASE.openshift.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+```
+
+## Creating a spreadsheet of carry commits from the previous release
```
Comment (tnozicka) [inline REBASE.openshift.md:68]: Would it all now be just this now?
```
git log $( git merge-base openshift/master v1.19.2 )..openshift/master --pretty=format:',%H,%s,https://github.com/openshift/kubernetes/commit/%H' | grep -v 'Merge pull request' | sed 's#,UPSTREAM: \([0-9]*\)\(:.*\)#,UPSTREAM: \1\2,https://github.com/kubernetes/kubernetes/pull/\1#'
```

File: REBASE.openshift.md (hunk 2 above)
Comment (tnozicka) [reply REBASE.openshift.md:65]: scratch that, we need patches to be applied as the latest, I got lost there.

Comment (marun) [reply REBASE.openshift.md:65]: Yeah, you've rediscovered my experience from experimenting with this as part of the transition. Using strategy 'ours' is the only reliable way to ensure that we're able to pick the carry commits cleanly. Anything else involves manual conflict resolution that risks introducing problems for no real benefit.

File: REBASE.openshift.md (hunk 1 above)
Comment (marun) [reply REBASE.openshift.md:30]: Done

File: REBASE.openshift.md (hunk 3 above)
Comment (marun) [reply REBASE.openshift.md:68]: Good call. Done.

**Pull request merged**